File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        if(cost.length == 1){return cost[0];}
        int[] costAtP = new int[cost.length];
        costAtP[0] = cost[0];
        costAtP[1] = cost[1];
        for(int i=2; i<cost.length; i++){
            costAtP[i] = Math.min(costAtP[i-1], costAtP[i-2])+cost[i];
        }
        return Math.min(costAtP[cost.length-1], costAtP[cost.length-2]);
    }
}

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
    let dp = [], len = cost.length;
    dp[0] = cost[0];
    dp[1] = cost[1];
    for(var i=2; i<len; i++){
        dp[i] = Math.min(dp[i-1], dp[i-2]) + cost[i];
    }
    return Math.min(dp[len-1], dp[len-2]);
};